### PR TITLE
Adding logic for testing yum-metadata copy logic

### LINKED ISSUE #91: Assert then YumMetadataFile copy save its new storage_path.
> Migrated from https://github.com/PulpQE/pulp-smash/issues/279
> author: @ipanova - https://github.com/ipanova
> date created: 2016-05-31T15:43:31Z

1. create  repo A
2. sync from a feed or upload yum_metadata_file
3. create repo B
4. copy yum_metadata_file from repo A to B
5. publish repo B
6. remove yum_metadata_file from repo A
7. remove orphans
8. publish repo B again ---> publish should succeed

for more info check https://pulp.plan.io/issues/1944
https://pulp.plan.io/issues/1935#note-5
https://github.com/pulp/pulp_rpm/pull/895

### PR DESCRIPTION
**Pull request by ragabala**
This test case will take care of the following
* It creates a repo with a feed from rpm with yum metadata
* It copies the repo to second repo
* It deletes the metadata content from repo1
* And then finally verifies whether the metadata content in second repo
is not affected

Refer [pulp.plan #1944 ](https://pulp.plan.io/issues/1944)
closes #91

### CODE REVIEW

File: pulp_2_tests/tests/rpm/api_v2/test_copy.py [hunk 1]
```diff
@@ -135,3 +142,91 @@ def test_all(self):
...
+
+        * `Pulp #1944 <https://pulp.plan.io/issues/1944>`_
+        * `Pulp-2-Tests #91
+          <https://github.com/PulpQE/Pulp-2-Tests/issues/91>`_
+
+        Do the following:
+
+        1. Create and sync a repository containing
+           ``yum_repo_metadata_file``.
+        2. Create another repo  and copy yum metadata from
```
Comment (bherrin3) [inline pulp_2_tests/tests/rpm/api_v2/test_copy.py]: double space

Comment (bherrin3) [inline pulp_2_tests/tests/rpm/api_v2/test_copy.py]:
```suggestion
        2. Create another repo and copy yum metadata from
```

File: pulp_2_tests/constants.py [hunk 2]
```diff
@@ -643,3 +643,7 @@
 
 Built from :data:`SRPM_UNSIGNED_FEED_URL` and :data:`SRPM`.
 """
+
+RPM_YUM_METADATA_FILE = 'https://repos.fedorapeople.org/' \
```
Comment (kersommoura) [inline pulp_2_tests/constants.py]: We have been avoid this kind of break line in code basis.  Just add a flag to make the linter happy.

File: pulp_2_tests/tests/rpm/api_v2/test_copy.py [hunk 3]
```diff
@@ -15,11 +15,18 @@
 )
 
 from pulp_2_tests.constants import (
+    RPM_NAMESPACES,
     RPM_SIGNED_URL,
     RPM_UNSIGNED_FEED_URL,
     RPM_UPDATED_INFO_FEED_URL,
+    RPM_YUM_METADATA_FILE,
+
```
Comment (kersommoura) [inline pulp_2_tests/tests/rpm/api_v2/test_copy.py]: This extra line can be removed.

File: pulp_2_tests/tests/rpm/api_v2/test_copy.py [hunk 4]
```diff
@@ -135,3 +142,91 @@ def test_all(self):
...
+        Do the following:
+
+        1. Create and sync a repository containing
+           ``yum_repo_metadata_file``.
+        2. Create another repo and copy yum metadata from
+           first repo to second repo.
+        3. Publish repo 2.
+        4. Remove the metadata units from the first repo. Delete
+           orphan nodes.
```
Comment (kersommoura) [inline pulp_2_tests/tests/rpm/api_v2/test_copy.py]: What about use packages or rpms? instead of nodes.

File: pulp_2_tests/constants.py (hunk 2 above)
Comment (kersommoura) [reply pulp_2_tests/constants.py]: What about add a note to inform this repo was not generated by pulp-fixtures?
```rst
.. NOTE:: This repository is not generated by `pulp-fixtures`_.
.. _pulp-fixtures: https://repos.fedorapeople.org/pulp/pulp/fixtures/
```

**Comment by ragabala** [comment]
@kersommoura,

Thanks for the review comments. Have made those changes. Lemme know if this works.

**Pull request merged**